docs/DeveloperGuide.md

```diff
@@ -1,34 +1,478 @@
+        * <a href ="#Tdeadline">Deadline Class</a> <br>
+        * <a href ="#Tevent">Event Class</a> <br>
+    * <a href ="#flashcard">Flashcard Class</a> <br>
+    * <a href ="#subject">Subject Class</a> <br>
+    * <a href ="#topic">Topic Class</a> <br>
+* <a href ="#command">Command Package</a> <br>
+* <a href ="#list">List Package</a> <br>
+* <a href ="#parser">Parser Package</a> <br>
+* <a href ="#ui">Ui Package</a> <br>
+* <a href ="#storage">Storage Package</a> <br>
+   - <a href ="#Sbuilder">Storage Builder</a> <br>
+   - <a href ="#Sstorage">Result</a> <br>
+* <a href ="#scope">Product Scope</a> <br>
+   - <a href ="#Ptarget">Target User Profile</a> <br>
+   - <a href ="#Pvalue">Value Proposition</a> <br>
+* <a href ="#user">User Stories</a> <br>
+* <a href ="#nfr">Non functional Requirements</a> <br>
+* <a href ="#glossary">Glossary</a> <br>
+* <a href ="#manual">Testing</a> <br>
+    - <a href ="#jartest">Testing the Jar File</a> <br>
+    - <a href ="#sourcetest">Testing the Source Code</a> <br>
+* <a href ="#develop">Development</a> <br>
+    - <a href ="#log">Logging</a> <br>
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Introduction <a name="intro"> </a>
 
+RevisED is a command line application to help students revise by creating flashcards and taking quizzes, 
+and keeping track of their deadlines, tasks and any other important dates. This application allows users to add subjects, 
+their different topics and create different flashcards in form of question and answers on each topic. Students can also 
+take quizzes which asks them the questions that they added in form of flashcards and prompted to answer them. 
+The student can view results of these quizzes immediately and are stored so that students can view them later. 
+Additionally, students can delete, and list subjects, topics, and flashcards and find and enter subjects and topics.   
 
-## Product scope
-### Target user profile
+This developer guide is for developers who wish to understand and/or develop ReviseED further. 
+This guide includes design, implementation, product scope, user stories, non-functional requirements, glossary, 
+testing and development sections to help developers better understand the application.
 
-{Describe the target user profile}
+## Design & implementation <a name= "design"> </a>
 
-### Value proposition
+This section describes how RevisED has been designed and implemented so that developers can better understand the 
+designing of the code, with the help of visual aids.
 
-{Describe the value proposition: what problem does it solve?}
+The application has been broken down into different segments so that users would find it easy to navigate from one segment 
+to the next one. There are three segments within the application:
+* Main: This segment edits the list of all the subjects present within the application and takes care of relevant commands 
```

Comment:
It might be better for you guys to have a diagram for you overall architecture so that the readers can know which classes or interfaces link with/implement each other, instead of trying to understand the overall architecture through the individual sequence/class diagrams below